### PR TITLE
Feature/tailor add edit design

### PR DESCRIPTION
**Pull request by syntialai**

### CODE REVIEW

File: features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/core/model/request/DesignSizeDetailRequest.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.future.tailormade_design_detail.core.model.request
+
+data class DesignSizeDetailRequest(
+    
+    val chest: Double,
```
Comment (blasiusneri) [inline features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/core/model/request/DesignSizeDetailRequest.kt]: float must be enough for size

File: features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddColorBottomSheetFragment.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import com.future.tailormade_design_detail.databinding.FragmentAddColorBottomSheetBinding
+import com.skydoves.colorpickerview.ColorEnvelope
+import com.skydoves.colorpickerview.listeners.ColorEnvelopeListener
+
+class AddColorBottomSheetFragment : BaseBottomSheetDialogFragment() {
+
+  companion object {
+    fun newInstance(onSubmitListener: (String, String) -> Unit, colorName: String? = null,
+        color: String? = null) = AddColorBottomSheetFragment().apply {
+      this.onSubmitListener = onSubmitListener
```
Comment (blasiusneri) [inline features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddColorBottomSheetFragment.kt:23]: put value into bundle

File: features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddColorBottomSheetFragment.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+  private fun getColorName() = binding.editTextColorName.text.toString()
+
+  private fun setColorPickerColor(color: String) {
+    val colour = Color.parseColor(color)
+    binding.viewColorPicker.setInitialColor(colour)
+  }
+
+  private fun setupColorPickerView() {
+    with(binding) {
```
Comment (blasiusneri) [inline features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddColorBottomSheetFragment.kt]: should be with(binding. viewColorPicker) {
}

File: features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddOrEditDesignFragment.kt [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+  }
+
+  private fun validate() {
+    with(binding) {
+      val name = editTextDesignName.text()
+      val price = editTextDesignPrice.text()
+      val discount = editTextDesignDiscount.text()
+      val description = editTextDesignDescription.text()
+
+      if (name.isNotBlank() && price.isNotBlank() && discount.isNotBlank() && description.isNotBlank() && viewModel.isPriceValid(
```
Comment (blasiusneri) [inline features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddOrEditDesignFragment.kt]: put this 
name.isNotBlank() && price.isNotBlank() && discount.isNotBlank() && description.isNotBlank() && viewModel.isPriceValid(price, discount)
in new method e.g. 
fun isPriveValid(price: Double, discount: Double) = name.isNotBlank() && price.isNotBlank() && discount.isNotBlank() && description.isNotBlank() && ...

File: features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddSizeBottomSheetFragment.kt [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+import com.future.tailormade.util.extension.text
+import com.future.tailormade_design_detail.core.model.ui.SizeDetailUiModel
+import com.future.tailormade_design_detail.databinding.FragmentAddSizeBottomSheetBinding
+
+class AddSizeBottomSheetFragment : BaseBottomSheetDialogFragment() {
+
+  companion object {
+    fun newInstance(onSubmitListener: (String, SizeDetailUiModel) -> Unit, name: String? = null,
+        sizeDetail: SizeDetailUiModel? = null) = AddSizeBottomSheetFragment().apply {
+      this.onSubmitListener = onSubmitListener
```
Comment (blasiusneri) [inline features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/addOrEditDesign/view/AddSizeBottomSheetFragment.kt:21]: put value in bundle

File: features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/designDetail/view/DesignDetailFragment.kt [hunk 6]
```diff
@@ -43,13 +43,20 @@ class DesignDetailFragment : BaseFragment() {
...
+  override fun getLogName() =
+      "com.future.tailormade_design_detail.feature.designDetail.view.DesignDetailFragment"
 
   override fun getViewModel(): BaseViewModel = viewModel
 
   override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
       savedInstanceState: Bundle?): View {
     binding = FragmentDesignDetailBinding.inflate(inflater, container, false)
+    binding.layoutDesignDetailGeneralInfo.buttonEditDesignDetail.setOnClickListener {
+      viewModel.designDetailResponse.value?.let { designDetailResponse ->
```
Comment (blasiusneri) [inline features/tailormade_design_detail/src/main/java/com/future/tailormade_design_detail/feature/designDetail/view/DesignDetailFragment.kt]: observe on onActivityCreated

File: tailor-app/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+<resources>
+    <string name="app_name">TailorMade\'s Tailor</string>
+    <string name="your_design_label">Your Design</string>
+    <string name="add_label">Add</string>
+    <string name="dashboard_title_state">Hello, tailor!</string>
+    <string name="dashboard_description_state">Let’s start adding your designs to attract customer by tapping the button above !</string>
+    <string name="delete_menu_label">Delete</string>
+    <string name="select_all_label">Select All</string>
+    <string name="delete_design_alert_dialog_title">Are you sure?</string>
+    <string name="delete_design_alert_dialog_content">You’ll delete %d design(s)!</string>
```
Comment (blasiusneri) [inline tailor-app/src/main/res/values/strings.xml:10]: please check plurals string for reference

**Review by blasiusneri — CHANGES_REQUESTED** [review]
please check commented code

File: tailor-app/src/main/res/values/strings.xml (hunk 7 above)
Comment (syntialai) [reply tailor-app/src/main/res/values/strings.xml:10]: Will fixed in dashboard branch kak

**Review by blasiusneri — APPROVED** [review]

**Pull request merged**